src/SimulationRunner.hh

```diff
@@ -160,9 +180,29 @@ namespace ignition
 
       /// \brief Add system after the simulation runner has been instantiated
       /// \note This actually adds system to a queue. The system is added to the
-      /// runner at the begining of the a simulation cycle (call to Run)
+      /// runner at the begining of the a simulation cycle (call to Run). It is
+      /// also responsible for calling `Configure` on the system.
+      /// \param[in] _system SystemPluginPtr to be added
+      /// \param[in] _entity Entity that system is attached to. If nullopt,
+      /// system is attached to a world.
+      /// \param[in] _sdf Pointer to the SDF of the entity. Nullopt defaults to
+      /// SDF of the entire world.
+      public: void AddSystem(const SystemPluginPtr &_system,
+          std::optional<Entity> _entity = std::nullopt,
+          std::optional<sdf::ElementPtr> _sdf = std::nullopt);
+
+      /// \brief Add system after the simulation runner has been instantiated
+      /// \note This actually adds system to a queue. The system is added to the
+      /// runner at the begining of the a simulation cycle (call to Run). It is
+      /// also responsible for calling `Configure` on the system.
       /// \param[in] _system System to be added
-      public: void AddSystem(const SystemPluginPtr &_system);
+      /// \param[in] _entity Entity of system to be added. Nullopt if system
+      /// doesn't connect to an entity.
+      /// \param[in] _sdf Pointer to the SDF of the entity. Nullopt defaults to
+      /// world.
+      public: void AddSystem(System *_system,
```

Comment:
Have you considered using a `std::shared_ptr<System>` here?

Reply:
Good call, I don't remember why I originally thought a raw pointer would make more sense. Changed in d8c6047